### PR TITLE
Showcase how to integrate with Dagger

### PR DESCRIPTION
**Pull request by sourishkrout**
This is incomplete, however, I'd like to get this branch merged since it includes a substantial refactor of some Kernel parts which go stale quickly. Merging `main` into this PR was kind of a pain.

### CODE REVIEW

**Comment by sourishkrout** [comment]
@pastuxso review this PR. It's incomplete but needs to be merge to not go stale. Foremost, if you could test the AWS profile switching stuff. Just want to be sure it'd survived my three-way merge.

**Review by pastuxso — APPROVED** [review]
LGTM 🚀 , the AWS Profile switch and SSH is working fine.

File: src/client/components/dagger.ts [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+  private onAction(a: DaggerStateAction) {
+    return (e: Event) => {
+      if (e.defaultPrevented) {
+        e.preventDefault()
+      }
+
+      let command = `${a.command} ${a.action}`
+      if (a.action === 'terminal') {
+        command = command.replace(' --progress=plain', '')
```
[the remark this answers is not in the document]
Comment (sourishkrout) [reply src/client/components/dagger.ts:128]: Yes

File: dagger/Pipeline.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  version: v3
+---
+
+# Express Dagger Pipelines in Notebooks
+
+One function's output is being used as another function's input. This is a common pattern to compose pipelines.
+
+In this notebook, we will explore how to express such pipelines using Dagger inside a literal notebook environment. As an example, we build and bundle this VS Code extension.
+
+> 💡 This demo is using a "slightly" modified version of the dagger binary based on https://github.com/dagger/dagger/pull/7479.
```
[the remark this answers is not in the document]
Comment (sourishkrout) [reply dagger/Pipeline.md:14]: That's fine. I'm going to rewrite those paragraphs upon finishing the work.

**Pull request merged**